Source/Moq.Analyzers/CallbackSignatureShouldMatchMockedMethodCodeFix.cs

```diff
@@ -60,10 +60,12 @@ private async Task<Document> FixCallbackSignatureAsync(SyntaxNode root, Document
 
         Debug.Assert(semanticModel != null, nameof(semanticModel) + " != null");
 
+#pragma warning disable S2583 // Conditionally executed code should be reachable
```

Comment:
Why suppress these? I'm not sure of the actual scenario, but based on the nullability annotations, it seems that a null `SemanticModel` is possible. Instead, should these be an early return without the `Debug.Assert`?

Reply:
I think this is actually a bug with the analyzer. It's theoretically reachable since the `document.GetSemanticModelAsync` can return a `SemanticModel?`, though I don't understand under what circumstance we could not be reachable. I have the assert in there to trip when that would be the case, but I can't actually get it to trip without doing some weird things in the harness.

Reply:
Only times where `SemanticModel` would be null are if the document does not support syntax tree or the project does not support compilation.

Reply:
I'm on my phone, but I have a feeling the Debug.Assert is what's tripping up the analyzer. If you remove it, does the analyzer behave correctly?

Reply:
It is; however, it's broken because it trips even in release mode where that won't fire. The examples in their documentation are also not this case https://rules.sonarsource.com/csharp/RSPEC-2583/.

In the case of DEBUG, I _want_ it to fire so I know when we trip over it.

I can put a comment in for _future maintainer_ to know why this is the way it is

Reply:
But why do you want to know when it fires? What do you plan on doing about it? I think it's a valid scenario where the only thing we can reasonably do is early return.

I think the debug asserts should just be removed entirely. Let me know if I'm misunderstanding the scenario / use case.

Reply:
Removed in db75e30c